bowtie/_benchmarks.py

```diff
@@ -420,6 +475,26 @@ def serializable(self):
         )
         return as_dict
 
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> BenchmarkMetadata:
+        return cls(
```

Comment:
This is doing too much manual `dict` access -- it should be doing `**data` (so that extra fields cause errors), and then the only thing it should do is coerce types of things in `data` which don't start as the right type.